### PR TITLE
Per user connection and channel limits definitions and types

### PR DESCRIPTION
**Pull request by Ayanda-D**
Part of: https://github.com/rabbitmq/rabbitmq-server/pull/2380

**This work has be carried out by** [Erlang Solutions](https://www.erlang-solutions.com/) **and sponsored by** [CloudAMQP](https://www.cloudamqp.com/)

### CODE REVIEW

**Comment by michaelklishin** [comment]
@Ayanda-D this PR cannot be accepted as is as it will break mixed version clusters. Should we expect a follow-up from ESL that makes the user record versioned? We can finish this work ourselves but this can take a while.

**Comment by Ayanda-D** [comment]
@michaelklishin hello, understood and thanks for the feedback. Some delays on our side, ESL have have now commenced work on this PR. We'll be pushing code updates later this week for mixed version cluster compatibility 👍

File: include/rabbit.hrl [hunk 1]
```diff
@@ -243,3 +261,7 @@
 %% Store metadata in the trace files when message tracing is enabled.
 -define(LG_INFO(Info), is_pid(whereis(lg)) andalso (lg ! Info)).
 -define(LG_PROCESS_TYPE(Type), ?LG_INFO(#{process_type => Type})).
+
+%% Execution timeout of connection and channel tracking operations
+-define(TRACKING_EXECUTION_TIMEOUT,
+        rabbit_misc:get_env(rabbit, tracking_execution_timeout, 5000)).
```
Comment (michaelklishin) [inline include/rabbit.hrl:267]: We have seen too many times before that the 5 second timeout is too prone to false positives on a heavily loaded system.

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
This has been merged manually with a merge commit. Keep this in mind when backporting.

**Comment by michaelklishin** [comment]
Backported to `v3.8.x`.

**Pull request closed without merging**